src/prefect/context.py

```diff
@@ -641,14 +640,6 @@ def root_settings_context():
         active_name = os.environ["PREFECT_PROFILE"]
         profile_source = "by environment variable"
 
-    if (
-        sys.argv[0].endswith("/prefect")
-        and len(sys.argv) >= 3
-        and sys.argv[1] == "--profile"
-    ):
-        active_name = sys.argv[2]
-        profile_source = "by command line argument"
```

Comment:
Looks like this was moved to the new setting source, but I think we need it here too to ensure we're populating the `profile` on the `SettingsContext` correctly. Maybe we need a utility to return the active profile so that we can reuse it? Longer term, I think we'll be able to take the profile off of the `SettingsContext`.